### DISCUSSION TITLE
not able to install pip in wsl ubuntu

### DISCUSSION THREAD

**Discussion by rajeshpotlia1** [microsoft/WSL#8579] [General]
When I try the code $ sudo apt install python3-pip , I get this error and it fails to install pip.
Please help me out.
![Screenshot (10)](https://user-images.githubusercontent.com/94539948/176411480-e447246b-c274-4524-b424-96f1dd15df80.png)
![Screenshot (9)](https://user-images.githubusercontent.com/94539948/176411469-07a450c8-8a06-4ee7-9749-15d7de1204d9.png)

**Comment by NotTheDr01ds** [comment]
While this isn't Stack Overflow, I'd still suggest reading the (applicable here as well), [post on why you shouldn't post text output as images](https://meta.stackoverflow.com/questions/285551/why-should-i-not-upload-images-of-code-data-errors-when-asking-a-question/285557#285557).  It's quite easy to copy output from WSL to the clipboard with `clip.exe`.  For example:

```
sudo apt install python3-pip 2>&1 | tee >(clip.exe)`
```

That will redirect all output and error to the Windows clipboard under WSL.  You can then paste it here in a code block (surrounded by triple backticks above and below).

**Comment by NotTheDr01ds** [comment]
As for why it isn't working, it's tough to read in it's current form (which is why I recommend pasting as text), but my guess is that you didn't `sudo apt update` before attempting to install pip.  Package versions are updated quite frequently in all Linux distributions, and you need to update the repo information before attempting to install.
